pkg/provision/engine.go

```diff
@@ -867,10 +868,44 @@ func (e *NativeEngine) uninstallDeployment(ctx context.Context, dl *gridtypes.De
 
 }
 
+func getZmountSize(wl *gridtypes.WorkloadWithID) (gridtypes.Unit, error) {
+	dataI, err := wl.WorkloadData()
+	if err != nil {
+		return 0, err
+	}
+	zmount, ok := dataI.(*zos.ZMount)
+	if !ok {
+		return 0, fmt.Errorf("failed to get workload data as a zmount '%v'", dataI)
+	}
+	return zmount.Size, nil
+}
+
+func sortZmountWorkloads(workloads []*gridtypes.WorkloadWithID) {
+	sort.Slice(workloads, func(i, j int) bool {
+		sizeI, err := getZmountSize(workloads[i])
+		if err != nil {
+			log.Error().Err(err).Stringer("id", workloads[i].ID).Msg("failed to get zmount size")
```

Comment:
i get why you log but i think it's not needed here. I won't show an error in the node logs because of wrong user data. This can confuse operators to see node giving errors that might make them think something wrong with the node itself while it's actual just user data that is wrong and that it was handled. 

I would personally not log it.